cli_tools/common/utils/param/param_utils.go

```diff
@@ -52,38 +53,61 @@ func PopulateMissingParameters(project *string, zone *string, region *string,
 		return err
 	}
 
+	scratchBucketRegion, err := populateBucket(scratchBucketGcsPath, zone, mgce, scratchBucketCreator,
+		file, project, storageClient)
+	if err != nil {
+		return err
+	}
+
+	if *zone == "" {
+		if aZone, err := zoneRetriever.GetZone(scratchBucketRegion, *project); err == nil {
+			*zone = aZone
+		} else {
+			return err
+		}
+	}
+
+	if err := PopulateRegion(region, *zone); err != nil {
+		return err
+	}
+	return nil
+}
+
+func populateBucket(scratchBucketGcsPath *string, zone *string, mgce domain.MetadataGCEInterface,
+	scratchBucketCreator domain.ScratchBucketCreatorInterface, file string, project *string,
+	storageClient domain.StorageClientInterface) (string, error) {
+
 	scratchBucketRegion := ""
 	if *scratchBucketGcsPath == "" {
-		scratchBucketName, sbr, err := scratchBucketCreator.CreateScratchBucket(file, *project)
+		fallbackZone := *zone
+		if fallbackZone == "" && mgce.OnGCE() {
+			var err error
+			if fallbackZone, err = mgce.Zone(); err != nil {
+				// reset fallback zone if failed to get zone from running GCE
+				fallbackZone = ""
+			}
+		}
+
+		scratchBucketName, sbr, err := scratchBucketCreator.CreateScratchBucket(file, *project, fallbackZone)
 		scratchBucketRegion = sbr
 		if err != nil {
-			return err
+			return "", err
 		}
 
 		*scratchBucketGcsPath = fmt.Sprintf("gs://%v/", scratchBucketName)
 	} else {
```

Comment:
This else block is actually validation of scratchBucketGcsPath (not population) and can be extracted to it's own func.

Then, the whole if/else could be moved to the calling func (PopulateMissingParameters) and this func (populateBucket) could just contain the if block.

Reply:
Moving if/else logic to calling func makes that func more complex. Ideally we can split populate to populateIfMissing + validate, but scratchBucketRegion needs to be determined here as well. Let's consider restructure this code section later since I prefer a conservative fix for now - there are many corner cases.